fs/config/ui.go

```diff
@@ -380,6 +382,33 @@ func backendConfig(ctx context.Context, name string, m configmap.Mapper, ri *fs.
 			if out.Option.Default == nil {
 				out.Option.Default = ""
 			}
+			if out.Option.Name == "endpoint" {
```

Comment:
Can you factor the contents of this out to a separate function please.

Reply:
Sure, I have set the quirks it needs and refactor the contents of this out to a separate function, please review again. 

Thanks for your time.